tests/src/Test/Tasty/Clash.hs

```diff
@@ -137,7 +140,7 @@ allTargets = [VHDL, Verilog, SystemVerilog]
 instance Default TestOptions where
   def =
     TestOptions
-      { hdlSim=True
+      { hdlSim=[GHDL, ModelSim, Vivado]
```

Comment:
It is no longer running simulation in Verilator because of this default. It's still running the `make` step but not the `sim` step. Before this PR:
```
$ cabal run clash-testsuite -- -p DivZero -j8
. 
  tests
    shouldwork
      Basic
        DivZero
          VHDL
            clash (gen):                OK (1.60s)
            GHDL (import):              OK
            ghdl (make testBench):      OK (0.55s)
            ghdl (sim testBench):       OK
          Verilog
            clash (gen):                OK (1.69s)
            iverilog (make testBench):  OK
            iverilog (sim testBench):   OK
            verilator (make testBench): OK (0.95s)
            verilator (sim testBench):  OK
          SystemVerilog
            clash (gen):                OK (1.63s)
            modelsim (vlib):            OK (0.03s)
            modelsim (vlog):            OK (0.08s)
            modelsim (sim testBench):   OK (0.60s)
            verilator (make testBench): OK (1.09s)
            verilator (sim testBench):  OK

All 15 tests passed (2.74s)
```
after this PR:
```
$ cabal run clash-testsuite -- -p DivZero -j8
. 
  tests
    shouldwork
      Basic
        DivZero
          VHDL
            clash (gen):                OK (1.76s)
            GHDL (import):              OK (0.02s)
            ghdl (make testBench):      OK (0.72s)
            ghdl (sim testBench):       OK
            Vivado:                     OK (21.37s)
          Verilog
            clash (gen):                OK (1.73s)
            iverilog (make testBench):  OK
            iverilog (sim testBench):   OK
            verilator (make testBench): OK (1.25s)
            Vivado:                     OK (21.40s)
          SystemVerilog
            clash (gen):                OK (1.69s)
            modelsim (vlib):            OK (0.02s)
            modelsim (vlog):            OK (0.06s)
            modelsim (sim testBench):   OK (0.75s)
            verilator (make testBench): OK (1.22s)

All 15 tests passed (23.15s)
```

Reply:
I also think the
```haskell
-- | Specify whether a test will use a traditional simulator, verilator, or
-- both when executing. By default both are used, but this can be changed when
-- a test is only relevant with one type of tool.
data Verilate = SimAndVerilate | VerilateOnly | SimOnly
```
functionality could be folded into `hdlSim`.